pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -628,9 +635,70 @@ func (r *Reconciler) computeDataHashAnnotation(existing, secret *v1.Secret) stri
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, opts controller.Options) error {
 	r.recorder = mgr.GetEventRecorderFor("external-secrets")
 
+	// Index .Spec.Target.Name to reconcile ExternalSecrets effectively when secrets have changed
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &esv1beta1.ExternalSecret{}, externalSecretSecretNameKey, func(obj client.Object) []string {
+		es := obj.(*esv1beta1.ExternalSecret)
+
+		if name := es.Spec.Target.Name; name != "" {
+			return []string{name}
+		}
+		return []string{es.Name}
+	}); err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(opts).
 		For(&esv1beta1.ExternalSecret{}).
-		Owns(&v1.Secret{}, builder.OnlyMetadata).
+		// Cannot use Owns since the controller does not set owner reference when creation policy is not Owner
+		Watches(
+			&v1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(r.findObjectsForSecret),
+			builder.WithPredicates(predicate.Funcs{
+				CreateFunc: func(event.TypedCreateEvent[client.Object]) bool {
+					return true
+				},
+				DeleteFunc: func(event.TypedDeleteEvent[client.Object]) bool {
+					return true
+				},
+				UpdateFunc: func(e event.TypedUpdateEvent[client.Object]) bool {
+					if !maps.Equal(e.ObjectNew.GetAnnotations(), e.ObjectOld.GetAnnotations()) {
+						return true
+					}
+					if !maps.Equal(e.ObjectNew.GetLabels(), e.ObjectOld.GetLabels()) {
+						return true
+					}
+					if e.ObjectNew.GetLabels()[esv1beta1.AnnotationDataHash] != "" && e.ObjectOld.GetLabels()[esv1beta1.AnnotationDataHash] != "" {
+						return e.ObjectNew.GetLabels()[esv1beta1.AnnotationDataHash] != e.ObjectOld.GetLabels()[esv1beta1.AnnotationDataHash]
+					}
```

Comment:
If someone manually updates a key that external-secrets populates, wouldn't this prevent that manual update from being passed through?

As in, if `e.ObjectNew.GetLabels()[esv1beta1.AnnotationDataHash] == e.ObjectOld.GetLabels()[esv1beta1.AnnotationDataHash]`, is this enough to be sure that the secret does not need to be reconciled at? For me  this would be the case of the resourceVersion comparison below 👇

Reply:
I've seen the tests cover this behavior I described above, I'm not entirely sure why. For me this condition would trigger a `false` to the need to reconcile on this update 🤔

Reply:
> If someone manually updates a key that external-secrets populates, wouldn't this prevent that manual update from being passed through?

Great catch! I haven't thought of a manual update, but you are right. I'll use `predicate.ResourceVersionChangedPredicate` instead 👍